test/MintableCappedERC20.js

```diff
@@ -1,104 +1,78 @@
 'use strict';
 
 const chai = require('chai');
+const { ethers } = require('hardhat');
 const {
-  utils: { splitSignature },
+    utils: { splitSignature },
 } = require('ethers');
-const { deployContract, MockProvider, solidity } = require('ethereum-waffle');
-chai.use(solidity);
 const { expect } = chai;
 
 const CHAIN_ID = 1;
 const ADDRESS_ZERO = '0x0000000000000000000000000000000000000000';
 
-const MintableCappedERC20 = require('../build/MintableCappedERC20.json');
-
-const { it } = require('mocha');
-
 describe('MintableCappedERC20', () => {
-  const [ownerWallet, userWallet] = new MockProvider().getWallets();
-  let token;
+    let owner;
+    let user;
+    let token;
 
-  beforeEach(async () => {
-    const name = 'test';
-    const symbol = 'test';
-    const decimals = 16;
-    const capacity = 0;
+    beforeEach(async () => {
+        [owner, user] = await ethers.getSigners();
 
-    token = await deployContract(ownerWallet, MintableCappedERC20, [
-      name,
-      symbol,
-      decimals,
-      capacity,
-    ]);
+        const mintableCappedERC20Factory = await ethers.getContractFactory('MintableCappedERC20', owner);
 
-    const amount = 1000000;
-    await token.mint(userWallet.address, amount);
-  });
+        token = await mintableCappedERC20Factory.deploy('test', 'test', 16, 0).then((d) => d.deployed());
 
-  describe('burnFrom', () => {
-    it('should burnFrom address after approval', async () => {
-      const issuer = userWallet.address;
-      const spender = ownerWallet.address;
-      const amount = 1000;
+        await token.mint(user.address, 1000000);
+    });
 
-      await expect(await token.connect(userWallet).approve(spender, amount))
-        .to.emit(token, 'Approval')
-        .withArgs(issuer, spender, amount);
+    describe('burnFrom', () => {
+        it('should burnFrom address after approval', async () => {
+            const amount = 1000;
 
-      await expect(await token.burnFrom(issuer, amount))
-        .to.emit(token, 'Transfer')
-        .withArgs(issuer, ADDRESS_ZERO, amount);
+            await expect(await token.connect(user).approve(owner.address, amount))
```

Comment:
```suggestion
            await expect(token.connect(user).approve(owner.address, amount))
```
the inner `await` is not needed.

Reply:
I've deleted a bunch of them